src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -228,4 +237,12 @@ public String getMessageCategories() {
     }
     return messageCategories;
   }
+  
+  public int getNumWords(String text) {
+    // use trim and split here to count the number of words in the text
```

Comment:
Add the following line to the beginning of the method. 

Objects.checkIfNull(text, "null text parameter");

This will ensure the method fails fast without revealing implementation details. When making new methods always make spec for it, making sure to keep in mind method modularity and what you want to promise the client. Here I think we should promise the client that the method will fail if they pass a null String. Another option could be returning 0, but my main point is that the client should now what will happen
 
Objects.checkIfNull(text, "null text parameter");

Reply:
I understand your thought. However, I might not be understanding what you are saying here... I put this but it doesn't recognize object. When I import the suggested library, it doesn't recognize requireNonNull

Reply:
@faesalm  the only thing unresolved